onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -2207,160 +2213,74 @@ Status SequentialPlanner::CreatePlan(
 
 #ifdef ORT_ENABLE_STREAM
 
-InlinedHashMap<std::string, IGraphPartitioner::GraphPartitioningStrategy>
-    IGraphPartitioner::name_type_map = {{std::string{"DeviceBasedPartitioner"}, GraphPartitioningStrategy::DeviceBasedPartition}};
-
 class DeviceBasedPartitioner : public IGraphPartitioner {
  public:
-  DeviceBasedPartitioner(const logging::Logger& logger, const std::string& configuration_file) : IGraphPartitioner(logger, configuration_file) {
+  DeviceBasedPartitioner(const logging::Logger& logger,
+                     const std::basic_string<PATH_CHAR_TYPE>& config_file) : IGraphPartitioner(logger, config_file) {
     Initialize();
   }
+
   ~DeviceBasedPartitioner() {
     if (need_dump_) {
-      DumpPartition();
+      if (!DumpPartition()) {
+        LOGS(logger_, WARNING) << "Failed to save partition info";
+      }
     }
   }
-  void DumpPartition() const;
-  Status PartitionGraph(const onnxruntime::GraphViewer& graph_viewer, const ExecutionProviders& execution_providers, std::vector<InlinedVector<NodeIndex>>& stream_nodes, ExecutionOrder execution_order) override;
-  virtual const std::string& Name() const override {
-    return name;
-  }
+
+  bool DumpPartition() const;
+
+  Status PartitionGraph(const onnxruntime::GraphViewer& graph_viewer,
+                        const ExecutionProviders& execution_providers,
+                        std::vector<InlinedVector<NodeIndex>>& stream_nodes,
+                        ExecutionOrder execution_order) override;
+
+  const char* Type() const override { return "DeviceBasedPartitioner"; }
```

Comment:
nit: use enum for Type?

Reply:
Enum is for sure better - planning on adding more partitioners using different algorithms, will extend to Enum shortly on that change.